StereoKit/Math/Matrix.cs

```diff
@@ -11,6 +11,12 @@ namespace StereoKit
 	/// together all sorts of transforms into a single Matrix! A Matrix
 	/// transform really shines when applied to many positions, as the more
 	/// expensive operations get cached within the matrix values.
+	///
+	/// Multiple matrix transforms can be combined by multiplying them, but note
+	/// that the order is different from the usual math order (and some other 3D 
```

Comment:
You're right that both conventions are more or less opposite sides of the same thing. I did not mean to imply there was anything *wrong* with SK in this respect. However, I *do* think that in math references column-major is the predominant form (e.g. check the forms in https://en.wikipedia.org/wiki/Rotation_matrix and https://mathworld.wolfram.com/RotationMatrix.html or the many 3D graphics books out there), hence the way I wrote it in the commit, to avoid surprises.

The crucial difference is the multiplication order that results from having to pre-multiply versus post-multiply. I **was** actually surprised I could use the "natural" order in SK, e.g. A*B*C to transform a point first by A, then by B, and C. In a lot of other 3D systems you need to remember that it's actually C*(B*(A*p)) that's happening under the hood and so need to push the first transform (A) *last*. The reference you linked summarizes this point nicely:

> From these observations, we can see that any series of transformations applied to a point or a vector when a row-major convention is used can be written in sequential order (or reading order). Imagine for instance that you want to translate point P with matrix T then rotate it around the z-axis with Rz then around the y-axis with Ry. You can write:
>
> P′=P∗T∗Rz∗Ry
>
> If you were to use a column-major notation you would need to call the transform in reverse order (which one might find counter-intuitive):
>
> P′=Ry∗Rz∗T∗P
> 
>So you may think, "there must be a reason to prefer one system to another". In fact, both conventions are correct and give us the same result, but for some technical reasons, Maths and Physics texts generally treat vectors as column vectors.

As listed there Blender and OpenGL are both column-major and so post-multiply M by a vector v. Those 2 systems, together with graphics math references, are the ones I've been exposed to ever since I started in 3D graphics, so I'm probably biased in that respect. This could therefore simply be a D3D versus OpenGL thing that we're encountering here.

All in all, my main goal was to make the multiplication order clear :wink:

Reply:
Yeah, I definitely appreciate it, and I think the clarification is great! I guess my nit here is the reference to 'usual math order', which does kinda indicate something unusual. It may or may not be usual depending on the person's background, and I think it'd be better to instead highlight that the ordering is due to SK using row-major, and then perhaps warn that some users may be more accustomed to column-major?